app/components/provider_interface/application_timeline_component.rb

```diff
@@ -27,18 +27,39 @@ def render?
 
   private
 
-    def events
-      changes = FindStatusChangeAudits.new(application_choice: application_choice).call.reverse
+    def status_change_events
+      changes = FindStatusChangeAudits.new(application_choice: application_choice).call
       changes = changes.select { |change| TITLES.has_key?(change.status) }
       changes.map do |change|
         Event.new(
```

Comment:
What would you think about making this polymorphic now that we have two two kinds of Timeline Entry? We could make a component for each type of entry, and have `events` return an array of `new`'d up components we call `render` on, eg

```ruby
> timeline.events
=> [StatusChangeEventComponent, StatusChangeEventComponent, NoteEventComponent]

timeline.events.each do |e|
  render(e)
end
```

where eg StatusChangeEventComponent looks like this:

```ruby
class StatusChangeEventComponent
  TITLES = {massive hash only relevant to this kind of entry}

  def title
    # deal with the audit log
  end

  def actor
    # deal with the audit log
  end

  def date
    # deal with the audit log
  end
end
```

then audit log concerns remain focussed — we could find a suitable way to reuse them if we ever needed to

Reply:
I think this is a great idea but will group these changes together with the attribution tweaks you and Adam has suggested in another PR.